문자열-내-p와-y의-개수.js

```diff
@@ -0,0 +1,20 @@
+const REG = /[py]/gi;
+
+function numPY(s){
+  var result = true;
+
+let pCount = 0;
+  let yCount = 0;
+
+  s.match(REG).forEach(chr => {
+    chr.toLowerCase() === 'y' ?
+      yCount++ :
+        pCount++;
+  });
+
+  return yCount === pCount;
+}
+
+
+console.log( numPY("pPoooyY"))
+console.log( numPY("Pyy"))
```

Comment:
오 ㅋㅋㅋ 정규 표현식까지 쓰시면서 푸셨네요 
코드 보시면 알겠지만 변수 result를 선언해 놓고 사용하지 않았어요
당연히 사용하지 않는 변수는 지워주는게 좋겠죠?

또한 var변수 선언은 전역적으로 호이스팅이 된다는 장점이 있지만 그에반해 변수 재할당이 되기 때문에 요즘은 var 선언을 하지않고 let, const 선언으로 사용하고 있어요
vat let const에 대해 찾아보시고 공부해 보시면 좋을것 같습니당

아래에는 다른 풀이방법이니 참고해 주시면 좋을 것 같아요

```suggestion
function solution(s) {
  let answer = true;
  let pCount = 0;
  let yCount = 0;

  for (let i in s) {
    if (s[i] == "p" || s[i] == "P") {
      pCount++;
    } else if (s[i] == "y" || s[i] == "Y") {
      yCount++;
    }
  }
  if (pCount != yCount) {
    answer = false;
  }
  return answer;
}

```